pilot/common/formatting.py

```diff
@@ -14,9 +14,9 @@ def check_unused_args(
         kwargs: Mapping[str, Any],
     ) -> None:
         """Check to see if extra parameters are passed."""
-        extra = set(kwargs).difference(used_args)
-        if extra:
-            raise KeyError(extra)
+        for item_arg in used_args:
+            if item_arg not in set(kwargs):
```

Comment:
This will have an impact on the global prompt template field verification. Can you consider the `template_is_strict` field of `PromptTemplate` to configure the prompt template to only take effect in a knowledge scene?